interface/components/SendTx/ContextDescription.js

```diff
@@ -0,0 +1,139 @@
+import React, { Component } from 'react';
+
+class ContextDescription extends Component {
+  formattedBalance() {
+    return EthTools.formatBalance(
+      web3.utils.toBN(this.props.value || 0),
+      '0,0.00[0000000000000000]',
+      'ether'
+    );
+  }
+
+  calculateTransferValue() {
+    const { value, etherPriceUSD } = this.props;
+
+    if (!value || !etherPriceUSD) {
+      return;
+    }
+
+    const bigValue = web3.utils.isHex(value)
+      ? new BigNumber(web3.utils.hexToNumberString(value))
+      : new BigNumber(value);
+    const fee = bigValue
+      .times(etherPriceUSD)
+      .dividedBy(new BigNumber('1000000000000000000'));
+    return this.formatter.format(fee);
+  }
+
+  determineTxType() {
+    if (this.props.isNewContract) return 'newContract';
+    if (this.props.toIsContract) {
+      if (this.props.executionFunction === 'transfer(address,uint256)') {
+        return 'tokenTransfer';
+      } else {
+        return 'genericFunctionExecution';
+      }
+    }
+    return 'etherTransfer';
+  }
+
+  renderNewContractDescription() {
+    const bytesCount = encodeURI(this.props.data).split(/%..|./).length - 1;
+
+    return (
+      <div className="context-description__sentence">
+        <div>
+          Upload <span className="bold">New Contract</span>
+        </div>
+        <div className="context-description__subtext">
+          About {bytesCount} bytes
+        </div>
+      </div>
+    );
+  }
+
+  renderTokenTransferDescription() {
+    if (this.props.params.length === 0) return;
+
+    const tokenCount = this.props.params[1].value.slice(
+      0,
+      -Math.abs(this.props.token.decimals)
+    );
```

Comment:
The statement above does not work for fractions of tokens. Consider these test cases:
```js
5e18.toString().slice(0, -18)
// expected: '5'
// output: '5'

0.5e18.toString().slice(0, -18)
// expected: '0.5'
// output: ''
```

Reply:
Good catch. Added a formatter/util function with some unit tests.

<img width="619" alt="screen shot 2018-09-30 at 11 04 28 pm" src="https://user-images.githubusercontent.com/3621728/46270550-418bf600-c505-11e8-9347-5f2135acbe1c.png">

Reply:
yay, thanks for making the formatter and a  test suite for it.